### ISSUE TITLE
Compile error.  "‘createGVNPass’ is not a member of ‘llvm’"

### ISSUE THREAD

**Issue by nshigit** [cmu-db/peloton#701]
[  1%] Original libpg makefile target
[  1%] Built target libpg_query
[  3%] Built target peloton-proto
[  3%] Building CXX object src/CMakeFiles/peloton.dir/codegen/code_context.cpp.o
/home/nshi/workspace/peloton/src/codegen/code_context.cpp: In constructor ‘peloton::codegen::CodeContext::CodeContext()’:
/home/nshi/workspace/peloton/src/codegen/code_context.cpp:72:25: error: ‘createGVNPass’ is not a member of ‘llvm’
   opt_pass_manager_.add(llvm::createGVNPass());
                         ^
src/CMakeFiles/peloton.dir/build.make:2561: recipe for target 'src/CMakeFiles/peloton.dir/codegen/code_context.cpp.o' failed
make[2]: *** [src/CMakeFiles/peloton.dir/codegen/code_context.cpp.o] Error 1
CMakeFiles/Makefile2:1146: recipe for target 'src/CMakeFiles/peloton.dir/all' failed
make[1]: *** [src/CMakeFiles/peloton.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by pmenon** [comment]
Which version of LLVM are you using (`llvm-config --version`). You should be using 3.7.

**Comment by nshigit** [comment]
OS: Linux mars 3.16.0-4-amd64 #1 SMP Debian 3.16.39-1 (2016-12-30) x86_64 GNU/Linux
llvm-3.9, clang-3.9

**Comment by pmenon** [comment]
3.9 might be too new. Please use 3.7.

**Comment by seojungmin** [comment]
I think that we could consider to investigate what should be done to support LLVM 3.8 or above.  If somebody would like to look into this matter, it would be grateful.  AFAIK, there is no certain reason that we should not support 3.8 or above.

**Comment by pmenon** [comment]
@seojungmin There isn't any reason we can't support newer versions of LLVM. I've tested most of the 3.x versions since these don't have breaking API changes. 4.x and 5.x are a different story.

I wanted to keeps things simple initially and stabilize everything on 3.7 before messing around with other versions. Having said that, I've fixed this in #697 (0912a8be8ecb9d66ca5c22707b2f6535ba515954) by adding some LLVM version-checking macros.
